File: cathode/src/main/java/net/simonvt/cathode/settings/LinkPromptBottomSheet.java
package net.simonvt.cathode.settings;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import net.simonvt.cathode.R;
import net.simonvt.cathode.settings.login.LoginActivity;

public class LinkPromptBottomSheet extends BottomSheetDialogFragment {

  public interface LinkPromptDismissListener {

    void onDismissLinkPrompt();
  }

  private LinkPromptDismissListener dismissListener;

  @Override public void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setCancelable(false);
    dismissListener = (LinkPromptDismissListener) requireActivity();
  }

  @Override public void onResume() {
    super.onResume();
    if (TraktLinkSettings.isLinkPrompted(requireContext())) {
      dismiss();
      dismissListener.onDismissLinkPrompt();
    }
  }

  @Nullable @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    return inflater.inflate(R.layout.prompt_link, container, false);
  }

  @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    view.findViewById(R.id.prompt_link_no).setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        Settings.get(requireContext())
            .edit()
            .putBoolean(TraktLinkSettings.TRAKT_LINK_PROMPTED, true)
            .apply();
        dismiss();
        dismissListener.onDismissLinkPrompt();
      }
    });
    view.findViewById(R.id.prompt_link_yes).setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        Intent i = new Intent(requireContext(), LoginActivity.class);
        startActivity(i);
      }
    });
  }
}

File: cathode/src/main/java/net/simonvt/cathode/settings/LinkPromptBottomSheet.kt
package net.simonvt.cathode.settings

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import net.simonvt.cathode.databinding.PromptLinkBinding
import net.simonvt.cathode.settings.login.LoginActivity

class LinkPromptBottomSheet : BottomSheetDialogFragment() {

  interface LinkPromptDismissListener {
    fun onDismissLinkPrompt()
  }

  private lateinit var dismissListener: LinkPromptDismissListener

  private var _binding: PromptLinkBinding? = null
  private val binding get() = _binding!!

  override fun onCreate(inState: Bundle?) {
    super.onCreate(inState)
    isCancelable = false
    dismissListener = requireActivity() as LinkPromptDismissListener
  }

  override fun onResume() {
    super.onResume()
    if (TraktLinkSettings.isLinkPrompted(requireContext())) {
      dismiss()
      dismissListener.onDismissLinkPrompt()
    }
  }

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    inState: Bundle?
  ): View? {
    _binding = PromptLinkBinding.inflate(inflater, container, false)
    return binding.root
  }

  override fun onViewCreated(view: View, inState: Bundle?) {
    super.onViewCreated(view, inState)
    binding.promptLinkNo.setOnClickListener {
      Settings.get(requireContext())
        .edit()
        .putBoolean(TraktLinkSettings.TRAKT_LINK_PROMPTED, true)
        .apply()
      dismiss()
      dismissListener.onDismissLinkPrompt()
    }
    binding.promptLinkYes
      .setOnClickListener {
        val i = Intent(requireContext(), LoginActivity::class.java)
        startActivity(i)
      }
  }
}
